### PR TITLE
Fix singleton

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ internal partial class FeatureFlag : IFeatureFlag
 
     private const string Prefix = "VSTEST_FEATURE_";
 
-    public static IFeatureFlag Instance => new FeatureFlag();
+    public static IFeatureFlag Instance { get; } = new FeatureFlag();
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs:17]:
```suggestion
    public static IFeatureFlag Instance { get; } ??= new FeatureFlag();
```

File: src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ internal partial class FeatureFlag : IFeatureFlag
 
     private const string Prefix = "VSTEST_FEATURE_";
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs:15]:
```suggestion
    private const string VSTEST_FEATURE = nameof(VSTEST_FEATURE);
```

Comment (nohwnd) [reply src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs:15]: And do the same for the rest of the env variables. And join them with `+ "_" +`.

Also the comment says "po*r*st-processing"

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/FeatureFlag/FeatureFlagTests.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+namespace Microsoft.TestPlatform.CoreUtilities.UnitTests.FeatureFlag;
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+
+[TestClass]
+public class FeatureFlagTests
+{
+    [TestMethod]
+    public void SingletonWorksAsExpected()
```
Comment (nohwnd) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/FeatureFlag/FeatureFlagTests.cs:13]:
```suggestion
    public void SingletonAlwaysReturnsTheSameInstance()
```

**Review by MarcoRossignoli — APPROVED** [review]

**Comment by MarcoRossignoli** [comment]
Thx @Evangelink missed that one, feature flag bag is immutable by design so it didn't fail any tests and also maybe singleton is not needed here, but let's keep the style, too sugars in this C# 😄

**Pull request merged**